src/solvers/refinement/string_refinement.cpp

```diff
@@ -336,6 +340,39 @@ static bool has_char_pointer_subtype(const typet &type, const namespacet &ns)
   return false;
 }
 
+/// \param type: a type
+/// \param ns: name space
+/// \return true if a subtype of `type` is string_typet.
+///         The meaning of "subtype" is in the algebraic datatype sense:
+///         for example, the subtypes of a struct are the types of its
+///         components, the subtype of a pointer is the type it points to,
+///         etc...
+static bool has_string_subtype(const typet &type, const namespacet &ns)
+{
+  if(type == string_typet())
```

Comment:
Can you pull the common code between this method and `has_char_pointer_subtype` into a separate method (`has_type_have_subtype(const typet & type, const namespacet &ns, const typet &required_subtype)`)